graphql/util.go

```diff
@@ -1163,3 +1164,67 @@ func interfaceToMap(ctx context.Context, data interface{}) (map[string]interface
 
 	return mapField, nil
 }
+
+type HasMatchingTasks struct {
+	Id       string
+	HasTasks bool
+}
+
+func concurrentlyBuildHasMatchingTasksMap(ctx context.Context, versions []model.Version, opts task.HasMatchingTasksOptions) (map[string]bool, error) {
+	wg := sync.WaitGroup{}
+	input := make(chan model.Version, len(versions))
```

Comment:
out of curiosity, why can't we just iterate through versions directly? (May be worth putting a comment so someone doesn't try to clean this up)

Reply:
In all of the examples I've seen, it's always been a channel — I think that looping through versions will get the correct result but it will also repeat work (so version A will get processed more than once, which would waste resources). I'll try to add a comment to the best of my understanding HAHA...

Reply:
Looping through the versions directly would mean that every worker processes every version, as opposed to dividing up the work.